### PR TITLE
Cleanup of TypeUtil and ContextHandler stop/start

### PR DESCRIPTION
**Pull request by gregw**
Hopefully non controversial changes from monster PR:
 + TypeUtil class shortname used more often and includes trailing digits
 + Fixed direct stopping/starting of a nested ContextHandler
 + Fixed null path handling in nested context
 + more tests for all of the above

### CODE REVIEW

**Comment by gregw** [comment]
@joakime @lachlan-roberts the CI failure is a java-19 flake.  Can you please review rather than wait for a green build.

File: jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ContextHandlerTest.java [hunk 1]
```diff
@@ -141,6 +141,44 @@ public void testSimpleGET() throws Exception
         assertThat(BufferUtil.toString(stream.getResponseContent()), equalTo(helloHandler.getMessage()));
     }
 
+    @Test
+    public void testNullPath() throws Exception
```
Comment (lachlan-roberts) [inline jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ContextHandlerTest.java:145]: I don't understand why we need `processMovedPermanently` and `processByContextHandler` both in `ContextHandler` one sends a 301 and other sends 302 response.

File: jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ContextHandler.java [hunk 2]
```diff
@@ -631,6 +637,57 @@ protected void doStart() throws Exception
         contextInitialized();
     }
 
+    @Override
```
Comment (janbartel) [inline jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ContextHandler.java:640]: FYI I think you've also fixed this and lines 277 297 in your other PR about changes to Handler.

File: jetty-ee9/jetty-ee9-webapp/src/test/java/org/eclipse/jetty/ee9/webapp/WebAppContextTest.java [hunk 3]
```diff
@@ -490,7 +490,7 @@ public void testBaseResourceAbsolutePath() throws Exception
         // On Unix / Linux this should have no issue.
         // On Windows with fully qualified paths such as "E:\mybase\webapps\dump.war" the
         // resolution of the Resource can trigger various URI issues with the "E:" portion of the provided String.
-        context.setResourceBase(warPath.toString());
+        context.setBaseResourceAsString(warPath.toString());
```
Comment (janbartel) [inline jetty-ee9/jetty-ee9-webapp/src/test/java/org/eclipse/jetty/ee9/webapp/WebAppContextTest.java:493]: FYI again, I think this is also part of 2 other PRs.

**Review by janbartel — APPROVED** [review]
Looks fine to me

**Review by olamy — APPROVED** [review]

File: jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ContextHandlerTest.java (hunk 1 above)
Comment (gregw) [reply jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/ContextHandlerTest.java:145]: I don't think we do.  I've removed processByContextHandler in #8978

**Pull request merged**